frontend/components/data-table.js

```diff
@@ -0,0 +1,550 @@
+import Component from '../component.js';
+import Pagination from './pagination.js';
+import PaginationSelector from './pagination-selector.js';
+
+export default class DataTable extends Component {
+  constructor({ element, phones, columnConfig }) {
+    super({ element });
+
+    this._props = {
+      config: columnConfig,
+      defaultPhones: phones,
+      phonesLength: phones.length,
+      phones,
+    };
+
+    this._render();
+    this._init();
+  }
+
+  _init() {
+    const defaultPerPage = 5;
+    const { phonesLength } = this._props;
+
+    this._state = {
+      filterValue: '',
+      selectedFilter: 'all',
+      orderValue: 'up',
+      orderName: '',
+      mainCheckboxValue: false,
+      currentPage: 0,
+      totalItemsCount: this.getItemsCount(),
+      buttonsCount: Math.ceil(phonesLength / defaultPerPage),
```

Comment:
State should not store values that can be calculated
https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state